developer_setup.md

```diff
@@ -1,323 +1,213 @@
 
-* Can't install any gems during bundle install
+Yarn, Gulp and Webpack are required to compile JavaScript assets. NodeJS version 10.16.x is required. If your distribution doesn't ship NodeJS 10.x, you can install [nvm](https://github.com/nvm-sh/nvm) and follow the setup steps. Then install `yarn`, `gulp-cli` and `webpack`.
 
+```bash
+nvm install 10
+npm install -g yarn gulp-cli webpack
 ```
-# install all dependencies to a local path
-bundle install --path vendor/bundle
-```
 
-* Can't install `nokogiri-1.7.2` on a Mac
+## Install Ruby and Bundler
+
+A Ruby version manager is *strongly* recommended. Use any one of the following:
+
+* [rbenv](https://github.com/rbenv/rbenv)
+* [rvm](http://rvm.io/)
+* [chruby](https://github.com/postmodern/chruby)
 
+Using the Ruby version manager, install `ruby` >= 2.4.0 and < 2.6.0 and `bundler` >= 1.16.
+
+## Clone the project
+
+```bash
+git clone git@github.com:ManageIQ/manageiq.git
+cd manageiq
 ```
-# install dependencies
-brew install pkgconfig
-brew install libxml2
 
-# test the gem can be compiled with the right libxml2
-export PKG_CONFIG_PATH=/usr/local/opt/libxml2/lib/pkgconfig
-gem install nokogiri -v 1.7.2 -- --use-system-libraries
+## Configure Project Manually
+
+All the steps below are wrapped inside the `bin/setup` script, but they are outlined here to explain how to completely bootstrap a developer environment. If you've run PostgreSQL in a container, be sure to export the `DATABASE_URL` variable to connect to the container over TCP instead of a UNIX file socket.
 
-# configure bundle to do so
-bundle config build.nokogiri --use-system-libraries --with-xml2-include=$(brew --prefix libxml2)/include/libxml2
+```bash
+export DATABASE_URL='posgresql://localhost:5432' # optional, only necessary if PostgreSQL is running in a container
 ```
 
-* Can't install `sys-proctable` on a Mac - a package missing even after bundle install succeeded
+### Create neccessary config files
 
+In order to run the application locally, a few sample config files need to be copied into their appropriate locations.
+
+```bash
+[[ -f "certs/v2_key" ]] || cp "certs/v2_key.dev" "certs/v2_key"
+[[ -f "config/cable.yml" ]] || cp "config/cable.yml.sample" "config/cable.yml"
+[[ -f "config/database.yml" ]] || cp "config/database.pg.yml" "config/database.yml"
+mkdir -p log
 ```
-bundle config specific_platform true
+
+### Install Bundler and Gems
+
+---
+**NOTE**
+
+macOS requires platform specific Gems. Run `bundle config specific_platform true` before installing.
+
+CentOS 7 requires the `rh-postgresql10` SCL environment in order to compile the `pq` Gem's native extensions. Run `bundle install` inside the `scl` environment: `scl enable rh-postgresql10 'bundle install'`.
+
+---
+
+```bash
+gem install bundler --version 1.16 --conservative
+bundle config specific_platform true # optional for non-macOS platforms
 bundle install
+bundle exec rails update:ui
+bundle exec rails db:create
```

Comment:
These could be `rake evm:db:reset` instead

Reply:
I included the two separately because I was mirroring the tasks invoked by `bin/setup` verbatim. I'll take a look at what `evm:db:reset` does compared to these two.